bridges/ethMultiversX/bridgeExecutor.go

```diff
@@ -169,6 +185,47 @@ func (executor *bridgeExecutor) StoreBatchFromMultiversX(batch *bridgeCore.Trans
 	return nil
 }
 
+// addBatchSCMetadataMvx fetches the logs containing sc calls metadata for the current batch
+func (executor *bridgeExecutor) addBatchSCMetadataMvx(ctx context.Context, transfers *bridgeCore.TransferBatch) (*bridgeCore.TransferBatch, error) {
+	if transfers == nil {
+		return nil, ErrNilBatch
+	}
+
+	events, err := executor.multiversXClient.GetBatchSCMetadata(ctx, transfers)
```

Comment:
perhaps add these events to an intermediary map, where key is deposit nonce

this way, the internal loop from addMetadataToTransferMvx will not be needed, as you can pass only the events for the nonce you're looking for